alica_engine/src/engine/BasicBehaviour.cpp

```diff
@@ -27,97 +28,77 @@ namespace alica
  * @param name The name of the behaviour
  */
 BasicBehaviour::BasicBehaviour(const std::string& name)
-    : name(name)
-    , engine(nullptr)
-    , failure(false)
-    , success(false)
-    , callInit(true)
-    , started(true)
-    , _configuration(nullptr)
-    , msInterval(100)
-    , msDelayedStart(0)
+        : _name(name)
+        , _engine(nullptr)
+        , _failure(false)
+        , _success(false)
+        , _callInit(true)
+        , _started(true)
+        , _configuration(nullptr)
+        , _msInterval(100)
+        , _msDelayedStart(0)
+        , _running(false)
+        , _contextInRun(nullptr)
+        , _behaviourTrigger(nullptr)
+        , _runThread(nullptr)
+        , _context(nullptr)
 {
-    this->running = false;
-    this->timer = new supplementary::Timer(0, 0);
-    this->timer->registerCV(&this->runCV);
-    this->behaviourTrigger = nullptr;
-    this->runThread = new std::thread(&BasicBehaviour::runInternal, this);
 }
 
 BasicBehaviour::~BasicBehaviour()
 {
-    this->started = false;
-    this->runCV.notify_all();
-    this->timer->start();
-    this->runThread->join();
-    delete this->runThread;
-    delete this->timer;
-    if (behaviourTrigger != nullptr) {
-        delete this->behaviourTrigger;
+    _started = false;
+    _runCV.notify_all();
+    if (_runThread) {
+        _runThread->join();
+        delete _runThread;
     }
 }
 
-const std::string& BasicBehaviour::getName() const
-{
-    return this->name;
-}
-
-void BasicBehaviour::setName(const std::string& name)
+bool BasicBehaviour::isRunningInContext(const RunningPlan* rp) const
 {
-    this->name = name;
+    // we run in the context of rp if rp is the context in run or the context in run is null and context is rp
+    RunningPlan* curInRun;
+    {
+        std::lock_guard<std::mutex> lck(_runLoopMutex);
```

Comment:
don't need a lock to grab a pointer

Reply:
i think you are right and it is no longer needed. Changed.